From ffe2a80fb76ccdc1781f817f6bbc9a8aa919816e Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Mon, 12 May 2025 09:11:05 -0500
Subject: [PATCH] net: pcs: qcom-ipq9574: remove "neg_mode" argument from
 ipq_pcs_get_state

Since commit c6739623c91bb ("net: phylink: pass neg_mode into
.pcs_get_state() method"), the "neg_mode" parameter is part of the
argument list of .pcs_get_state(). This is available starting with
v6.14. However, we want to use the backported IPQ9574 driver on v6.12.
Remove this parameter from ipq_pcs_get_state(), as it is not part of
.pcs_get_state() in v6.12.

Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 drivers/net/pcs/pcs-qcom-ipq9574.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/pcs/pcs-qcom-ipq9574.c
+++ b/drivers/net/pcs/pcs-qcom-ipq9574.c
@@ -457,7 +457,7 @@ static void ipq_pcs_disable(struct phyli
 	clk_disable_unprepare(qpcs_mii->tx_clk);
 }
 
-static void ipq_pcs_get_state(struct phylink_pcs *pcs, unsigned int neg_mode,
+static void ipq_pcs_get_state(struct phylink_pcs *pcs,
 			      struct phylink_link_state *state)
 {
 	struct ipq_pcs_mii *qpcs_mii = phylink_pcs_to_qpcs_mii(pcs);
